Jasmine/Tetris/ViewModels/TetrisGameViewModel.swift

```diff
@@ -1,4 +1,219 @@
 import Foundation
 
+/// The main view model for Tetris game
 class TetrisGameViewModel {
+
+    weak var delegate: TetrisGameViewControllerDelegate?
+
+    fileprivate var tetrisGrid = TetrisGrid()
+
+    fileprivate(set) var upcomingTiles: [String] = []
+    fileprivate var fallingTileText: String?
+    fileprivate var landingCoordinate: Coordinate?
+
+    fileprivate(set) var currentScore: Int = 0 {
+        didSet {
+            delegate?.redisplay(newScore: currentScore)
+        }
+    }
+
+    let timer = CountDownTimer(totalTimeAllowed: Constants.Game.Tetris.totalTime)
+
+    private(set) var gameStatus = GameStatus.notStarted {
+        didSet {
+            delegate?.notifyGameStatusUpdated()
+        }
+    }
+
+    /// Populate upcomingTiles and set listeners for the timer
+    init() {
+        populateUpcomingTiles()
+        setTimerListener()
+    }
+
+    private func populateUpcomingTiles() {
+        for _ in 0..<Constants.Game.Tetris.upcomingTilesCount {
+            upcomingTiles.append(getRandomWord())
+        }
+    }
+
+    private func setTimerListener() {
```

Comment:
the implementation in Grid VM initiated the timer twice.
Btw most of your comments are in outdated files, think you weren't reviewing the latest.